src/StepFunctionGraph/GraphUtil.ts

```diff
@@ -182,7 +182,10 @@ export const adjustDepthMatrix = (matrix: number[][], graph: Digraph): number[][
                 );
 
                 accum = accum.concat(outDegree).filter(v => v !== nextVertex);
-                removalArray = removalArray.concat(outDegree).filter(v => v !== nextVertex);
+
```

Comment:
Nit: this function has grown a lot, and it's a little difficult to follow the logic. Could you break it down to a few functions and call the "high level algorithm" from this function to make it more readable?

Reply:
I agree it's long and hard to follow. It's easily the most complicated function in the renderer because it just requires that much data to get an x value. I'll keep the request in mind, but I'm not sure I see anywhere to break it up at a glance without passing in a lot of params. I've been considering breaking the way to get data from the drawn map and Digraph into a simpler function, which would free up a lot of the line bloat. I'll try to tackle this if I have time for it.

Reply:
There is also the getX function that could use some breaking out of the drawn and Digraph data retrieval which other functions can benefit from and reduce line bloat.